macros/get_metric_sql.sql

```diff
@@ -38,9 +74,13 @@ VALIDATION ROUND ONE - THE MACRO LEVEL!
     {%- do exceptions.raise_compiler_error("From v0.3.0 onwards, the where clause takes a single string, not a list of filters. Please fix to reflect this change") %}
 {% endif %}
 
-{% do metrics.validate_grain(grain, metric_tree['full_set'], metric_tree['base_set'])%}
+{% if not is_develop_macro %}
+    {% do metrics.validate_grain(grain, metric_tree['full_set'], metric_tree['base_set'])%}
+{% endif %}
 
-{% do metrics.validate_expression_metrics(metric_tree['full_set'])%}
+{% if not is_develop_macro %}
+    {% do metrics.validate_expression_metrics(metric_tree['full_set'])%}
+{% endif %}
```

Comment:
Are we skipping this validation because it's unnecessary given your validation above, or because it would fail? 

What would it take for this to not need to be skipped?

Reply:
We're skipping this one because expression metrics aren't supported in this first version of develop. They require an understanding of the metric dag that would be very tricky to rebuild with just the string input.

Reply:
OK so it would break in some unexpected way? That’s a bummer. 

Could `validate_grain` and `validate_expression_metrics` infer that if they're given a mediocre tree (i.e. just string input) that it's a develop context and just early exit without doing any validation?

Reply:
So I just merged #86 into this branch and that moved those two validate steps into the higher level macros. That way we can remove the conditional